Tactics/Common.lean

```diff
@@ -115,21 +118,54 @@ which was obtained by reducing:\n\t{e}"
 reduce an expression `e` (of type `BitVec w`) to be of the form `?n#w`,
 and then reflect `?n` to build the meta-level bitvector -/
 def reflectBitVecLiteral (w : Nat) (e : Expr) : MetaM (BitVec w) := do
-  if e.hasFVar then
+  if e.hasFVar || e.hasMVar then
     throwError "Expected a ground term, but {e} has free variables"
 
-  if let some ⟨n, x⟩ ← _root_.getBitVecValue? e then
-    if h : n = w then
-      return x.cast h
-    else
-      throwError "Expected a bitvector of width {w}, but\n\t{e}\nhas width {n}"
+  let some ⟨n, x⟩ ←_root_.getBitVecValue? e
+    | throwError "Failed to reflect:\n\t{e}\ninto a BitVec"
 
-  let x ← mkFreshExprMVar (Expr.const ``Nat [])
-  let e' ← mkAppM ``BitVec.ofNat #[toExpr w, x]
-  if (←isDefEq e e') then
-    return BitVec.ofNat w (← reflectNatLiteral x)
+  if h : n = w then
+    return x.cast h
   else
-    throwError "Failed to unify, expected:\n\t{e'}\nbut found:\n\t{e'}"
+    throwError "Expected a bitvector of width {w}, but\n\t{e}\nhas width {n}"
+
+
+/-! ## Hypothesis types -/
+namespace SymContext
+
+/-- `h_err_type state` returns an Expr for `r .ERR <state> = .None`,
+the expected type of `h_err` -/
+def h_err_type (state : Expr) : Expr :=
+  mkAppN (mkConst ``Eq [1]) #[
+    mkConst ``StateError,
+    mkApp2 (.const ``r []) (.const ``StateField.ERR []) state,
+    .const ``StateError.None []
+  ]
```

Comment:
```suggestion
/-- `h_err_type state` returns an Expr for `r .ERR <state> = .None`,
the expected type of `h_err` -/
def h_err_type (state : Expr) : Expr :=
  mkAppN (mkConst ``Eq [1]) #[
    mkConst ``StateError,
    mkApp2 (.const ``r []) (.const ``StateField.ERR []) state,
    .const ``StateError.None []
  ]
```

If I may be annoying, can we name this according to the term we build? So something like `h_State_error_eq_none`? In particular, this seems hard to audit when the definition changes.

I admit, a long name doesn't make me happy 😢

Reply:
Note that the various uses of `h_err_type` are pretty strongly tied together; if the expression we want at one of it's callsite changes we almost surely want the same change at all the other callsites. Does that alleviate your concerns re:auditability

Reply:
@alexkeizer @bollu What's the conclusion of this discussion? I'm happy with the way things stand right now. If you both are happy, hit "resolve" and we're ready to merge this PR!

Reply:
I'm happy with the current naming; I'd be a bit hesitant to adopt longer names like @bollu suggests, so I'd like to merge it as is, if Sid agrees

Reply:
I'm resolving this after a chat with @bollu